sdk/function_id.go

```diff
@@ -0,0 +1,89 @@
+package sdk
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+
+	"github.com/capeprivacy/cli/attest"
+	"github.com/capeprivacy/cli/crypto"
+	"github.com/capeprivacy/cli/entities"
+	"github.com/capeprivacy/cli/pcrs"
+)
+
+type FunctionIDRequest struct {
+	URL          string
+	FunctionName string
+	FunctionAuth entities.FunctionAuth
+	PcrSlice     []string
+
+	// For development use only: skips validating TLS certificate from the URL
+	Insecure bool
+}
+
+func GetFunctionID(functionReq FunctionIDRequest) (string, error) {
```

Comment:
As mentioned in the above comment, this isn't implemented by using a websocket w/ the enclave, you will just be making an HTTP GET request to a server.

You could set up a test server that returns a function ID (ie mimic what the supervisor is doing) and then develop your `GetFunctionID` function to call it ... Something like ...

```go
func testServer(t *testing.T) *httptest.Server {
	// mimic what supervisor does
	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if _, err := w.Write([]byte(`{"id": "abc123"`)); err != nil {
			t.Error(err)
		}
	}))
	
	t.Cleanup(s.Close)

	return s
}

func TestFunctionID(t *testing.T) {
	s := testServer(t) // get a "supervisor"
	id, err := GetFunctionID(FunctionIDRequest{URL: s.URL, FunctionName: "my-function"})
	if err != nil {
		t.Fatal(err)
	}

	if got, want := id, "abc123"; got != want {
		t.Fatalf("didn't get expected function id, got %s, wanted %s", got, want)
	}
}

```